prusti-interface/src/environment/borrowck/regions.rs

```diff
@@ -84,7 +84,11 @@ impl PlaceRegions {
                         not supported".to_string()
                     ));
                 }
-                x => unreachable!("{:?}", x),
+                x => {
+                    return Err(PlaceRegionsError::Unsupported(
+                        format!("determining the region of projection {:?} is not supported", x)
+                    ));
+                }
```

Comment:
So far we avoided dumping internal representations of the program in the "unsupported" error messages, to ensure that the messages are not too intimidating for the users. ("internal" error messages, by contrast, can be as ugly as we want.)

So, I think we should remove the `{:?}`. The message `"determining the region of a projection is not supported"` would be fine, but a bit too vague. The best thing would be to complete the match and add a separate error message for `Index`, `Downcast`, `ConstantIndex`, `Subslice` (e.g. `"determining the region of a downcast/array indexing/... is not supported"`).

Reply:
Updated and pushed